qemu/tests/virtio_fs_share_data.py

```diff
@@ -526,9 +528,20 @@ def viofs_svc_stop_start(action, cmd, expect_status):
                 if create_dir_winapi_cmd:
                     error_context.context("Create new directory with WinAPI's "
                                           "CreateDirectory.", test.log.info)
-                    s, o = session.cmd_status_output(create_dir_winapi_cmd)
+                    session.cmd(create_dir_winapi_cmd % fs_dest)
+                    start_time = time.time()
+                    end_time = start_time + 60
+
+                    while time.time() < end_time:
+                        s, o = session.cmd_status_output(check_winapi_dir_cmd % fs_dest)
+                        if s:
+                            test.log.debug("Dir isn't created by winapi after %s seconds."
+                                           % (time.time() - start_time))
+                        else:
+                            break
+                        time.sleep(1)
```

Comment:
I think this part could be implemented by the `utils_misc.wait_for`.
Maby we can have a try.

Reply:
Hi yongxue,first thanks for the suggestion.But after I checked utils_misc.wait_for(),it is to "Wait for the process to finish". While for this part, check_winapi_dir_cmd process can get the result very quickly, here I just want to run check_winapi_dir_cmd many times to check if the dir existed because session.cmd(create_dir_winapi_cmd % fs_dest) has no return code.